File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    int index = 0;
    int sum = 0;
    
    while(sum != progresses.size()) {
        int cur = progresses[index];
        int need = 100 - progresses[index];
        int needCount = need / speeds[index];
        if (need % speeds[index] != 0) needCount++;
        
        for(int i=0;i<progresses.size();i++) {
            progresses[i] += needCount * speeds[i];
        }
        
        int count = 0;
        
        for (int j=index;j<progresses.size();j++) {
            if (progresses[j] >= 100) {
                count++;
            }
            else {
                index = j;
                break;
            }
        }  
        
        if (count > 0) {
            answer.push_back(count);
            sum += count;
        }
        
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.io.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        List<Integer> temp = new ArrayList<>();
        
        int day = (100 - progresses[0] + speeds[0] - 1) / speeds[0];
        int cnt = 1;
        for (int i=1;i<progresses.length;++i) {
            progresses[i] += speeds[i] * day;
            
            if (progresses[i] >= 100) {
                cnt++;
            }
            else {
                int newDay = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
                day += newDay;
                temp.add(cnt);
                cnt = 1;
            }
            
        }
        
        if (cnt > 0) temp.add(cnt);
        answer = temp.stream().mapToInt(Integer::intValue).toArray();
        
        return answer;
    }
}